[Automation] javac compile fails for org.openjfx:javafx-base:13-ea+14

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25296069081/job/74155118918
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.openjfx:javafx-base:11.0.2
Raw versions JSON: [
  "13-ea+14",
  "13-ea+14b",
  "13.0.1",
  "13.0.2",
  "14",
  "14-ea+1",
  "14-ea+2",
  "14-ea+4",
  "14-ea+6",
  "14-ea+7",
  "14-ea+8",
  "14-ea+9",
  "14.0.1",
  "14.0.2.1",
  "14.0.2",
  "15",
  "15-ea+1",
  "15-ea+2",
  "15-ea+3",
  "15-ea+4",
  "15-ea+5",
  "15-ea+6",
  "15-ea+7",
  "15-ea+8",
  "15.0.1",
  "16",
  "16-ea+1",
  "16-ea+2",
  "16-ea+3",
  "16-ea+4"
]
Parsed version count: 30
Parsed versions: 13-ea+14 13-ea+14b 13.0.1 13.0.2 14 14-ea+1 14-ea+2 14-ea+4 14-ea+6 14-ea+7 14-ea+8 14-ea+9 14.0.1 14.0.2.1 14.0.2 15 15-ea+1 15-ea+2 15-ea+3 15-ea+4 15-ea+5 15-ea+6 15-ea+7 15-ea+8 15.0.1 16 16-ea+1 16-ea+2 16-ea+3 16-ea+4
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.openjfx:javafx-base:11.0.2:13-ea+14
========================================================================================
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage native-image run
========================================================================================
Starting stage 'native-image run' for 13-ea+14 at 2026-05-04T01:06:39Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean test -Pcoordinates="org.openjfx:javafx-base:11.0.2"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 6s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.openjfx:javafx-base:11.0.2
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 3s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.openjfx:javafx-base:11.0.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 11s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 13-ea+14 at 2026-05-04T01:06:50Z with exit code 1 after 11s
❌ ERROR: 'native-image run' for 13-ea+14 failed with exit code 1.
SOFT-FAIL[native-image run][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean test -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage javac compile
========================================================================================
Starting stage 'javac compile' for 13-ea+14 at 2026-05-04T01:06:50Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 887ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with 

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
--scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 878ms
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 13-ea+14 at 2026-05-04T01:06:53Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 13-ea+14 failed with exit code 1.
FAILED[javac compile][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25296069081/job/74155118913
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.openjfx:javafx-base:11.0.2
Raw versions JSON: [
  "13-ea+14",
  "13-ea+14b",
  "13.0.1",
  "13.0.2",
  "14",
  "14-ea+1",
  "14-ea+2",
  "14-ea+4",
  "14-ea+6",
  "14-ea+7",
  "14-ea+8",
  "14-ea+9",
  "14.0.1",
  "14.0.2.1",
  "14.0.2",
  "15",
  "15-ea+1",
  "15-ea+2",
  "15-ea+3",
  "15-ea+4",
  "15-ea+5",
  "15-ea+6",
  "15-ea+7",
  "15-ea+8",
  "15.0.1",
  "16",
  "16-ea+1",
  "16-ea+2",
  "16-ea+3",
  "16-ea+4"
]
Parsed version count: 30
Parsed versions: 13-ea+14 13-ea+14b 13.0.1 13.0.2 14 14-ea+1 14-ea+2 14-ea+4 14-ea+6 14-ea+7 14-ea+8 14-ea+9 14.0.1 14.0.2.1 14.0.2 15 15-ea+1 15-ea+2 15-ea+3 15-ea+4 15-ea+5 15-ea+6 15-ea+7 15-ea+8 15.0.1 16 16-ea+1 16-ea+2 16-ea+3 16-ea+4
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.openjfx:javafx-base:11.0.2:13-ea+14
========================================================================================
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage native-image run
========================================================================================
Starting stage 'native-image run' for 13-ea+14 at 2026-05-04T01:06:39Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean test -Pcoordinates="org.openjfx:javafx-base:11.0.2"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.openjfx:javafx-base:11.0.2
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

BUILD FAILED in 3s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.openjfx:javafx-base:11.0.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 12s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 13-ea+14 at 2026-05-04T01:06:51Z with exit code 1 after 12s
❌ ERROR: 'native-image run' for 13-ea+14 failed with exit code 1.
SOFT-FAIL[native-image run][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean test -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage javac compile
========================================================================================
Starting stage 'javac compile' for 13-ea+14 at 2026-05-04T01:06:51Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 866ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 891ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 13-ea+14 at 2026-05-04T01:06:54Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 13-ea+14 failed with exit code 1.
FAILED[javac compile][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25296069081/job/74155118925
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.openjfx:javafx-base:11.0.2
Raw versions JSON: [
  "13-ea+14",
  "13-ea+14b",
  "13.0.1",
  "13.0.2",
  "14",
  "14-ea+1",
  "14-ea+2",
  "14-ea+4",
  "14-ea+6",
  "14-ea+7",
  "14-ea+8",
  "14-ea+9",
  "14.0.1",
  "14.0.2.1",
  "14.0.2",
  "15",
  "15-ea+1",
  "15-ea+2",
  "15-ea+3",
  "15-ea+4",
  "15-ea+5",
  "15-ea+6",
  "15-ea+7",
  "15-ea+8",
  "15.0.1",
  "16",
  "16-ea+1",
  "16-ea+2",
  "16-ea+3",
  "16-ea+4"
]
Parsed version count: 30
Parsed versions: 13-ea+14 13-ea+14b 13.0.1 13.0.2 14 14-ea+1 14-ea+2 14-ea+4 14-ea+6 14-ea+7 14-ea+8 14-ea+9 14.0.1 14.0.2.1 14.0.2 15 15-ea+1 15-ea+2 15-ea+3 15-ea+4 15-ea+5 15-ea+6 15-ea+7 15-ea+8 15.0.1 16 16-ea+1 16-ea+2 16-ea+3 16-ea+4
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.openjfx:javafx-base:11.0.2:13-ea+14
========================================================================================
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage native-image run
========================================================================================
Starting stage 'native-image run' for 13-ea+14 at 2026-05-04T01:06:42Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="13-ea+14" ./gradlew clean test -Pcoordinates="org.openjfx:javafx-base:11.0.2"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.openjfx:javafx-base:11.0.2
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
13 actionable tasks: 5 executed, 8 up-to-date
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 3s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.openjfx:javafx-base:11.0.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 11s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 13-ea+14 at 2026-05-04T01:06:54Z with exit code 1 after 12s
❌ ERROR: 'native-image run' for 13-ea+14 failed with exit code 1.
SOFT-FAIL[native-image run][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="13-ea+14" ./gradlew clean test -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
========================================================================================
 org.openjfx:javafx-base:11.0.2:13-ea+14 stage javac compile
========================================================================================
Starting stage 'javac compile' for 13-ea+14 at 2026-05-04T01:06:54Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 897ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not find javafx-base-13-ea+14-linux.jar (org.openjfx:javafx-base:13-ea+14).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/13-ea+14/javafx-base-13-ea+14-linux.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 862ms
 UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.openjfx/javafx-base/11.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 13-ea+14 at 2026-05-04T01:06:57Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 13-ea+14 failed with exit code 1.
FAILED[javac compile][13-ea+14][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="13-ea+14" ./gradlew clean compileTestJava -Pcoordinates="org.openjfx:javafx-base:11.0.2"]
```

--- a/metadata/org.openjfx/javafx-base/13-ea/reachability-metadata.json
+++ b/metadata/org.openjfx/javafx-base/13-ea/reachability-metadata.json
@@ -2,7 +2,7 @@
   "reflection" : [
     {
       "condition" : {
-        "typeReached" : "javafx.beans.property.adapter.JavaBeanIntegerProperty"
+        "typeReached" : "com.sun.javafx.reflect.MethodUtil"
       },
       "type" : "com.sun.javafx.reflect.Trampoline",
       "methods" : [

--- a/metadata/org.openjfx/javafx-base/index.json
+++ b/metadata/org.openjfx/javafx-base/index.json
@@ -12,7 +12,8 @@
     ],
     "allowed-packages": [
       "org.openjfx",
-      "javafx.beans.property.adapter"
+      "javafx.beans.property.adapter",
+      "com.sun.javafx.reflect"
     ]
   },
   {

--- a/stats/org.openjfx/javafx-base/13-ea/execution-metrics.json
+++ b/stats/org.openjfx/javafx-base/13-ea/execution-metrics.json
@@ -105,5 +105,112 @@
       "test_file": "tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/PropertyDescriptorTest.java",
       "metadata_file": "metadata/org.openjfx/javafx-base/13-ea/reachability-metadata.json"
     }
+  },
+  "fix_javac_fail:2026-05-04": {
+    "timestamp": "2026-05-04T03:22:27.470209Z",
+    "library": "org.openjfx:javafx-base:13-ea",
+    "starting_commit": "b7d6f1a9dbf439dc0f38681fb7b43aea97703e1d",
+    "ending_commit": "afdd4a61759e662e55c6e7cf4cc1fb3d376f5968",
+    "previous_library": "org.openjfx:javafx-base:13-ea",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "13-ea",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 0.701754,
+            "coveredCalls": 40,
+            "totalCalls": 57
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 0.706897,
+        "coveredCalls": 41,
+        "totalCalls": 58
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 6721,
+          "missed": 56770,
+          "ratio": 0.105858,
+          "total": 63491
+        },
+        "line": {
+          "covered": 1769,
+          "missed": 12591,
+          "ratio": 0.123189,
+          "total": 14360
+        },
+        "method": {
+          "covered": 510,
+          "missed": 4069,
+          "ratio": 0.111378,
+          "total": 4579
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "13-ea",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 0.701754,
+            "coveredCalls": 40,
+            "totalCalls": 57
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 0.706897,
+        "coveredCalls": 41,
+        "totalCalls": 58
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 6721,
+          "missed": 56770,
+          "ratio": 0.105858,
+          "total": 63491
+        },
+        "line": {
+          "covered": 1769,
+          "missed": 12591,
+          "ratio": 0.123189,
+          "total": 14360
+        },
+        "method": {
+          "covered": 510,
+          "missed": 4069,
+          "ratio": 0.111378,
+          "total": 4579
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 231169,
+      "cached_input_tokens_used": 2229760,
+      "output_tokens_used": 23564,
+      "iterations": 5,
+      "cost_usd": 1.8218,
+      "tested_library_loc": 1769,
+      "metadata_entries": 2,
+      "previous_library_metadata_entries": 2,
+      "code_coverage_percent": 12.32,
+      "previous_library_coverage_percent": 12.32
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/FXCollectionsInnerCheckedObservableListTest.java",
+      "metadata_file": "metadata/org.openjfx/javafx-base/13-ea/reachability-metadata.json"
+    }
   }
 }

--- a/stats/org.openjfx/javafx-base/13-ea/stats.json
+++ b/stats/org.openjfx/javafx-base/13-ea/stats.json
@@ -4,8 +4,8 @@
     "dynamicAccess" : {
       "breakdown" : {
         "reflection" : {
-          "coverageRatio" : 0.263158,
-          "coveredCalls" : 15,
+          "coverageRatio" : 0.701754,
+          "coveredCalls" : 40,
           "totalCalls" : 57
         },
         "resources" : {
@@ -14,27 +14,27 @@
           "totalCalls" : 1
         }
       },
-      "coverageRatio" : 0.275862,
-      "coveredCalls" : 16,
+      "coverageRatio" : 0.706897,
+      "coveredCalls" : 41,
       "totalCalls" : 58
     },
     "libraryCoverage" : {
       "instruction" : {
-        "covered" : 5799,
-        "missed" : 57692,
-        "ratio" : 0.091336,
+        "covered" : 6721,
+        "missed" : 56770,
+        "ratio" : 0.105858,
         "total" : 63491
       },
       "line" : {
-        "covered" : 1551,
-        "missed" : 12809,
-        "ratio" : 0.108008,
+        "covered" : 1769,
+        "missed" : 12591,
+        "ratio" : 0.123189,
         "total" : 14360
       },
       "method" : {
-        "covered" : 460,
-        "missed" : 4119,
-        "ratio" : 0.100459,
+        "covered" : 510,
+        "missed" : 4069,
+        "ratio" : 0.111378,
         "total" : 4579
       }
     }

--- a/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/FXCollectionsInnerCheckedObservableListTest.java
+++ b/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/FXCollectionsInnerCheckedObservableListTest.java
@@ -1,0 +1,54 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_openjfx.javafx_base;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import org.junit.jupiter.api.Test;
+
+public class FXCollectionsInnerCheckedObservableListTest {
+
+    @Test
+    void addAllCollectionOverloadsCopyInputThroughTypedArrays() {
+        ObservableList<Integer> checkedValues = FXCollections.checkedObservableList(
+                FXCollections.observableArrayList(10, 40), Integer.class);
+
+        boolean appended = checkedValues.addAll(List.of(50, 60));
+        boolean inserted = checkedValues.addAll(1, List.of(20, 30));
+
+        assertThat(appended).isTrue();
+        assertThat(inserted).isTrue();
+        assertThat(checkedValues).containsExactly(10, 20, 30, 40, 50, 60);
+    }
+
+    @Test
+    void addAllVarargsCopiesInputThroughTypedArray() {
+        ObservableList<Integer> checkedValues = FXCollections.checkedObservableList(
+                FXCollections.observableArrayList(10), Integer.class);
+
+        boolean changed = checkedValues.addAll(20, 30);
+
+        assertThat(changed).isTrue();
+        assertThat(checkedValues).containsExactly(10, 20, 30);
+    }
+
+    @Test
+    void setAllOverloadsCopyInputThroughTypedArrays() {
+        ObservableList<Integer> checkedValues = FXCollections.checkedObservableList(
+                FXCollections.observableArrayList(10, 20), Integer.class);
+
+        boolean changedFromVarargs = checkedValues.setAll(30, 40);
+        boolean changedFromCollection = checkedValues.setAll(List.of(50, 60));
+
+        assertThat(changedFromVarargs).isTrue();
+        assertThat(changedFromCollection).isTrue();
+        assertThat(checkedValues).containsExactly(50, 60);
+    }
+}

--- a/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/MethodUtilTest.java
+++ b/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/MethodUtilTest.java
@@ -1,0 +1,89 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_openjfx.javafx_base;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.sun.javafx.reflect.MethodUtil;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+public class MethodUtilTest {
+
+    @Test
+    void publicLookupMethodsReturnInvocableBeanMethods() throws Exception {
+        Method getter = MethodUtil.getMethod(AccessibleBean.class, "getName", new Class<?>[0]);
+        Method setter = MethodUtil.getMethod(AccessibleBean.class, "setName", new Class<?>[] {String.class});
+        AccessibleBean bean = new AccessibleBean("Ada");
+
+        MethodUtil.invoke(setter, bean, new Object[] {"Grace"});
+        Object name = MethodUtil.invoke(getter, bean, null);
+        Method[] methods = MethodUtil.getMethods(AccessibleBean.class);
+
+        assertThat(name).isEqualTo("Grace");
+        assertThat(methods)
+                .extracting(Method::getName)
+                .contains("getName", "setName", "describe");
+    }
+
+    @Test
+    void publicMethodsFastPathEnumeratesAccessibleMethodsWithoutSecurityManager() throws Exception {
+        Method[] methods = invokeGetPublicMethods(AccessibleBean.class);
+
+        assertThat(methods)
+                .extracting(Method::getName)
+                .contains("getName", "setName", "describe");
+    }
+
+    @Test
+    void internalPublicMethodScanCollectsInterfaceAndImplementationMethods() throws Exception {
+        Map<Object, Method> signatures = new HashMap<>();
+        Method getInternalPublicMethods = MethodUtil.class.getDeclaredMethod(
+                "getInternalPublicMethods", Class.class, Map.class);
+        getInternalPublicMethods.setAccessible(true);
+
+        boolean complete = (boolean) getInternalPublicMethods.invoke(null, AccessibleBean.class, signatures);
+
+        assertThat(complete).isTrue();
+        assertThat(signatures.values())
+                .extracting(Method::getName)
+                .contains("getName", "setName", "describe");
+    }
+
+    private static Method[] invokeGetPublicMethods(Class<?> type) throws Exception {
+        Method getPublicMethods = MethodUtil.class.getDeclaredMethod("getPublicMethods", Class.class);
+        getPublicMethods.setAccessible(true);
+        return (Method[]) getPublicMethods.invoke(null, type);
+    }
+
+    public interface Described {
+        String describe();
+    }
+
+    public static final class AccessibleBean implements Described {
+        private String name;
+
+        public AccessibleBean(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public String describe() {
+            return "bean:" + name;
+        }
+    }
+}

--- a/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/PropertyReferenceTest.java
+++ b/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/PropertyReferenceTest.java
@@ -1,0 +1,83 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_openjfx.javafx_base;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.sun.javafx.property.PropertyReference;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import org.junit.jupiter.api.Test;
+
+public class PropertyReferenceTest {
+
+    @Test
+    void reflectsReadableWritablePropertyAndObservablePropertyGetter() {
+        PropertyReference<String> reference = new PropertyReference<>(StandardBean.class, "name");
+
+        assertThat(reference.isReadable()).isTrue();
+        assertThat(reference.isWritable()).isTrue();
+        assertThat(reference.hasProperty()).isTrue();
+        assertThat(reference.getType()).isEqualTo(String.class);
+    }
+
+    @Test
+    void reflectsBooleanPropertyUsingIsGetter() {
+        PropertyReference<Boolean> reference = new PropertyReference<>(BooleanBean.class, "active");
+
+        assertThat(reference.isReadable()).isTrue();
+        assertThat(reference.isWritable()).isTrue();
+        assertThat(reference.hasProperty()).isFalse();
+        assertThat(reference.getType()).isEqualTo(boolean.class);
+    }
+
+    @Test
+    void reflectsWriteOnlyPropertyByScanningPublicMethods() {
+        PropertyReference<Integer> reference = new PropertyReference<>(WriteOnlyBean.class, "score");
+
+        assertThat(reference.isReadable()).isFalse();
+        assertThat(reference.isWritable()).isTrue();
+        assertThat(reference.hasProperty()).isFalse();
+        assertThat(reference.getType()).isEqualTo(int.class);
+    }
+
+    public static final class StandardBean {
+        private final StringProperty name = new SimpleStringProperty(this, "name", "Ada");
+
+        public String getName() {
+            return name.get();
+        }
+
+        public void setName(String newName) {
+            name.set(newName);
+        }
+
+        public StringProperty nameProperty() {
+            return name;
+        }
+    }
+
+    public static final class BooleanBean {
+        private boolean active = true;
+
+        public boolean isActive() {
+            return active;
+        }
+
+        public void setActive(boolean active) {
+            this.active = active;
+        }
+    }
+
+    public static final class WriteOnlyBean {
+        private int score;
+
+        public void setScore(int score) {
+            this.score = score;
+        }
+    }
+}

--- a/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/ReadOnlyPropertyDescriptorTest.java
+++ b/tests/src/org.openjfx/javafx-base/13-ea/src/test/java/org_openjfx/javafx_base/ReadOnlyPropertyDescriptorTest.java
@@ -1,0 +1,189 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_openjfx.javafx_base;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyChangeSupport;
+import java.util.ArrayList;
+import java.util.List;
+import javafx.beans.property.adapter.ReadOnlyJavaBeanIntegerProperty;
+import javafx.beans.property.adapter.ReadOnlyJavaBeanIntegerPropertyBuilder;
+import org.junit.jupiter.api.Test;
+
+public class ReadOnlyPropertyDescriptorTest {
+
+    @Test
+    void readOnlyPropertyRegistersAndRemovesSpecificPropertyChangeListener() throws NoSuchMethodException {
+        SpecificPropertyChangeBean bean = new SpecificPropertyChangeBean(3);
+
+        ReadOnlyJavaBeanIntegerProperty property = ReadOnlyJavaBeanIntegerPropertyBuilder.create()
+                .bean(bean)
+                .name("score")
+                .build();
+        List<Object> invalidatedObservables = observeInvalidations(property);
+
+        assertThat(bean.getScore()).isEqualTo(3);
+        assertThat(bean.scoreListeners).hasSize(1);
+
+        bean.setScore(5);
+        assertThat(bean.getScore()).isEqualTo(5);
+        assertThat(invalidatedObservables).containsExactly(property);
+
+        property.dispose();
+        assertThat(bean.scoreListeners).isEmpty();
+    }
+
+    @Test
+    void readOnlyPropertyRegistersAndRemovesNamedPropertyChangeListener() throws NoSuchMethodException {
+        NamedPropertyChangeBean bean = new NamedPropertyChangeBean(7);
+
+        ReadOnlyJavaBeanIntegerProperty property = ReadOnlyJavaBeanIntegerPropertyBuilder.create()
+                .bean(bean)
+                .name("score")
+                .build();
+        List<Object> invalidatedObservables = observeInvalidations(property);
+
+        assertThat(bean.getScore()).isEqualTo(7);
+        assertThat(bean.addedPropertyNames).containsExactly("score");
+        assertThat(bean.propertyChangeListeners).hasSize(1);
+
+        bean.setScore(11);
+        assertThat(bean.getScore()).isEqualTo(11);
+        assertThat(invalidatedObservables).containsExactly(property);
+
+        property.dispose();
+        assertThat(bean.removedPropertyNames).containsExactly("score");
+        assertThat(bean.propertyChangeListeners).isEmpty();
+    }
+
+    @Test
+    void readOnlyPropertyRegistersAndRemovesGlobalPropertyChangeListener() throws NoSuchMethodException {
+        GlobalPropertyChangeBean bean = new GlobalPropertyChangeBean(13);
+
+        ReadOnlyJavaBeanIntegerProperty property = ReadOnlyJavaBeanIntegerPropertyBuilder.create()
+                .bean(bean)
+                .name("score")
+                .build();
+        List<Object> invalidatedObservables = observeInvalidations(property);
+
+        assertThat(bean.getScore()).isEqualTo(13);
+        assertThat(bean.propertyChangeListeners).hasSize(1);
+
+        bean.setScore(17);
+        assertThat(bean.getScore()).isEqualTo(17);
+        assertThat(invalidatedObservables).containsExactly(property);
+
+        property.dispose();
+        assertThat(bean.removedListeners).hasSize(1);
+        assertThat(bean.propertyChangeListeners).isEmpty();
+    }
+
+    private static List<Object> observeInvalidations(ReadOnlyJavaBeanIntegerProperty property) {
+        List<Object> invalidatedObservables = new ArrayList<>();
+        property.addListener(invalidatedObservables::add);
+        return invalidatedObservables;
+    }
+
+    public static final class SpecificPropertyChangeBean {
+        private final List<PropertyChangeListener> scoreListeners = new ArrayList<>();
+        private int score;
+
+        public SpecificPropertyChangeBean(int score) {
+            this.score = score;
+        }
+
+        public int getScore() {
+            return score;
+        }
+
+        public void setScore(int score) {
+            int oldScore = this.score;
+            this.score = score;
+            PropertyChangeEvent event = new PropertyChangeEvent(this, "score", oldScore, score);
+            for (PropertyChangeListener listener : List.copyOf(scoreListeners)) {
+                listener.propertyChange(event);
+            }
+        }
+
+        public void addScoreListener(PropertyChangeListener listener) {
+            scoreListeners.add(listener);
+        }
+
+        public void removeScoreListener(PropertyChangeListener listener) {
+            scoreListeners.remove(listener);
+        }
+    }
+
+    public static final class NamedPropertyChangeBean {
+        private final List<String> addedPropertyNames = new ArrayList<>();
+        private final List<String> removedPropertyNames = new ArrayList<>();
+        private final List<PropertyChangeListener> propertyChangeListeners = new ArrayList<>();
+        private int score;
+
+        public NamedPropertyChangeBean(int score) {
+            this.score = score;
+        }
+
+        public int getScore() {
+            return score;
+        }
+
+        public void setScore(int score) {
+            int oldScore = this.score;
+            this.score = score;
+            PropertyChangeEvent event = new PropertyChangeEvent(this, "score", oldScore, score);
+            for (PropertyChangeListener listener : List.copyOf(propertyChangeListeners)) {
+                listener.propertyChange(event);
+            }
+        }
+
+        public void addPropertyChangeListener(String propertyName, PropertyChangeListener listener) {
+            addedPropertyNames.add(propertyName);
+            propertyChangeListeners.add(listener);
+        }
+
+        public void removePropertyChangeListener(String propertyName, PropertyChangeListener listener) {
+            removedPropertyNames.add(propertyName);
+            propertyChangeListeners.remove(listener);
+        }
+    }
+
+    public static final class GlobalPropertyChangeBean {
+        private final PropertyChangeSupport changeSupport = new PropertyChangeSupport(this);
+        private final List<PropertyChangeListener> propertyChangeListeners = new ArrayList<>();
+        private final List<PropertyChangeListener> removedListeners = new ArrayList<>();
+        private int score;
+
+        public GlobalPropertyChangeBean(int score) {
+            this.score = score;
+        }
+
+        public int getScore() {
+            return score;
+        }
+
+        public void setScore(int score) {
+            int oldScore = this.score;
+            this.score = score;
+            changeSupport.firePropertyChange("score", oldScore, score);
+        }
+
+        public void addPropertyChangeListener(PropertyChangeListener listener) {
+            propertyChangeListeners.add(listener);
+            changeSupport.addPropertyChangeListener(listener);
+        }
+
+        public void removePropertyChangeListener(PropertyChangeListener listener) {
+            removedListeners.add(listener);
+            propertyChangeListeners.remove(listener);
+            changeSupport.removePropertyChangeListener(listener);
+        }
+    }
+}

